### ISSUE TITLE
Tilt input inverted in some devices.

### ISSUE THREAD

**Issue by blackman91** [hrydgard/ppsspp#8054]
I am using an Archos gamepad 2 Tablet and when I select tilt to control analog stick or dpad, the commands seems to be inverted between the X and Y axis.
Up = Left
Down = Right
Right = Down
Left = Up

Tried the calibrate option but it doesn't fix the issue. I have seen this weird behaviour in other devices like androi asus.

**Comment by unknownbrackets** [comment]
We pull in the data from the accelerometer.  This is the documented / correct coordinate system for them to be sent in:

http://developer.android.com/images/axis_device.png

We flip this to handle landscape mode.  Are you playing in portrait?  Possibly we should not flip when you switch to portrait mode.

Otherwise, your device probably is sending incorrect accelerometers.  Maybe we can include this in calibration by asking you to tilt it upwards and then leftwards, so we get the axis sign as well.

-[Unknown]

**Comment by blackman91** [comment]
I am playing in landscape, apparently even in landscape the axes are still detected as if I were playing in portrait. That idea about the calibration is great, hope it gets implemented.

**Comment by unknownbrackets** [comment]
@hrydgard do you think we should respect both display AND screen rotation?  I'm not sure what input types make the most sense.  Right now it's hardcoded as: `bool portrait = dp_yres > dp_xres;`.

-[Unknown]

**Comment by hrydgard** [comment]
Hm, yes, if the PSP display is rotated, that check should be rotated appropriately...  But if blackman simply means playing in regular landscape or portrait mode on his device, that check should take care of it.

**Comment by blackman91** [comment]
Tried the latest dev, playing on landscape but the tilt is still wrong like I described in my original post, moving the device up and down while in landscape moves the character left and right.

**Comment by blackman91** [comment]
Could you guys add an option to switch the axis? that I think would solve most of the problems with this issue.

**Comment by blackman91** [comment]
Do you know of any android apps to switch the tilt axis?

**Comment by unknownbrackets** [comment]
So I've tested this and for me, it feels right...

The on-screen controls are displayed on screen, and react in the same direction as they are displayed.  That is:

 * If I select screen rotation (under Settings), the controls are rotated too.
 * If I select display rotation (under Display layout editor), the controls are not rotated.

This is intentional, I think.  Which method of rotation are you using?  I think you should set display layout editor to landscape, and screen rotation to portrait.

-[Unknown]

**Comment by blackman91** [comment]
Thanks your workaround worked for me I had to set the screen rotation to portrait and the rotation in the layout editor to portrait reversed to make the tilt works as it supposed to, now using my device in landscape if I move right the character goes right:
http://i1312.photobucket.com/albums/t539/talos910/Screenshot_2016-02-08-02-28-45_zpsusboflmu.png

**Comment by hrydgard** [comment]
Yeah, I think your device has tilt reversed. We could provide some checkboxes to invert, would be more obvious than the described workaround...

**Comment by blackman91** [comment]
Thanks that would be awesome and would solve the issue on other devices with this weird tilt, the weird thing is android games that use the tilt like galaxy on fire 2 and racing games detect the tilt of my device just fine in landscape.

**Comment by blackman91** [comment]
The open source emu Mupen AE for android has a menu to assign the x axis or the y axis of the accelerometer to the x axis or y axis of the stick, that way even if your device axes are reversed you can still play like supposed to. You think that could be added to this emu? Also the emu Myboy detects the axes of my device correctly in landscape for some reason, maybe it has a a check to detect if the axes do not reverse when changing to landscape?

**Comment by ResurrectionRedmi9** [comment]
Tilt input really need some improvement as unknown mention here https://github.com/hrydgard/ppsspp/issues/14844#issuecomment-918413813
